pkg/pillar/cmd/vaultmgr/vaultmgr.go

```diff
@@ -332,17 +344,29 @@ func Run() {
 	}
 	defer logf.Close()
 
+	// Context to pass around
+	ctx := vaultMgrContext{}
+
+	// initialize publishing handles
+	initializeSelfPublishHandles(&ctx)
+
 	switch flag.Args()[0] {
 	case "setupVaults":
 		if err = setupFscryptEnv(); err != nil {
+			publishVaultStatus(&ctx, defaultImgVaultName)
```

Comment:
How is publish followed by log.Fatal useful?
The subscriber is not going to receive it since the publisher is gone.

Reply:
Perhaps you shouldn't Fatal?
Perhaps feed err into publish??

Reply:
Addressed in the latest patch.